src/lib/util/ChainGate.tsx

```diff
@@ -0,0 +1,24 @@
+import { FC, ReactNode } from 'react';
+import { Network } from '../constants/networks';
+
+interface ChainGateProps {
+	chainId: number;
+	children: ReactNode;
+}
+
+const ChainGate: FC<ChainGateProps> = ({ chainId, children }) => {
```

Comment:
I think this should be moved out of `util` folder. Probably in zUI or some other higher level

Reply:
For this I followed the same placement as zApp-BuyDomains and zApp-Staking. Maybe something we could discuss with the others on where to place?